lib/jsdom/living/nodes/HTMLFormElement-impl.js

```diff
@@ -66,18 +66,47 @@ class HTMLFormElementImpl extends HTMLElementImpl {
     if (!isConnected(this)) {
       return;
     }
-    const ev = this._ownerDocument.createEvent("HTMLEvents");
-    ev.initEvent("submit", true, true);
-    if (this.dispatchEvent(ev)) {
-      this.submit();
-    }
+
+    this.submit();
   }
 
   submit() {
+    const submitEvent = Event.createImpl(
+      [
+        "submit",
+        { bubbles: true, cancelable: true }
+      ],
+      { isTrusted: true }
+    );
+
+    if (!this.dispatchEvent(submitEvent)) {
+      return;
+    }
```

Comment:
I was thinking about the same thing. I moved event dispatching to `submit()/reset()` because having that in `_doSubmit / _doReset()` results in events not being dispatched when reset or submit is called directly.

I think that the method names may be a little confusing since `_doSubmit`  is being called from `HTMLInputElement` (https://github.com/jsdom/jsdom/blob/d5339dfb882ff2c03d7f5fa05cb842abc30a4efc/lib/jsdom/living/nodes/HTMLButtonElement-impl.js#L21). I decided to leave it that way because I wasn't sure about the the `isConnected(this)` check in those private methods and didn't want to cause any regression.